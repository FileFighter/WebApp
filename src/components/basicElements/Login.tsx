import React, {Dispatch, FormEvent, ReactElement, SetStateAction, useState} from 'react';
import {Button, Col, Container, Form, Image, Row, Spinner} from "react-bootstrap";
import {loginWithUsernameAndPassword} from "../../background/api/auth";

import logo from "../../assets/images/logos/logoWithWhiteBorder.png";

export interface LoginInputInterface {
    handleSubmit: (event: FormEvent) => void,
    username: string | number | string[] | undefined,
    setUsername: Dispatch<SetStateAction<string>>,
    password: string | number | string[] | undefined,
    setPassword: Dispatch<SetStateAction<string>>,
    isLoading: boolean,
    setIsLoading: Dispatch<SetStateAction<boolean>>,
    stayLoggedIn: boolean,
    setStayLoggedIn: Dispatch<SetStateAction<boolean>>,
    errorMessage: string | null
}

function Login(): ReactElement {
    const [userName, setUsername] = useState<string>("");
    const [password, setPassword] = useState<string>("");
    const [stayLoggedIn, setStayLoggedIn] = useState<boolean>(true);
    const [errorMessage, setErrorMessage] = useState<string>("");
    const [isLoading, setIsLoading] = useState<boolean>(false);


    const handleSubmit = (event: FormEvent) => {
        event.preventDefault();
        setIsLoading(true)
        loginWithUsernameAndPassword(userName, password, stayLoggedIn)
            .then(() => {
                //nothing to do here :)
                setErrorMessage("");
                setIsLoading(false);
            })
            .catch(((error) => {
                console.log(error);
                setIsLoading(false)
                setErrorMessage(error.response?.data.message);
            }))


    };

    return (
        <Container fluid className="h-100 ml-0 mr-0 login-page">
            <LoginInteractionArea
                handleSubmit={handleSubmit}
                username={userName}
                setUsername={setUsername}
                password={password}
                setPassword={setPassword}
                isLoading={isLoading}
                setIsLoading={setIsLoading}
                stayLoggedIn={stayLoggedIn}
                setStayLoggedIn={setStayLoggedIn}
                errorMessage={errorMessage}
            />
        </Container>);
}

export function LoginInteractionArea(props: LoginInputInterface) {
    const {
        handleSubmit,
        username,
        setUsername,
        password,
        setPassword,
        isLoading,
        setIsLoading,
        stayLoggedIn,
        setStayLoggedIn,
        errorMessage,
    } = props
    return (
        <div className="login-container pr-1 pl-1 mr-auto ml-auto">
            <LoginHeader/>
            <LoginInput
                handleSubmit={handleSubmit}
                username={username}
                setUsername={setUsername}
                password={password}
                setPassword={setPassword}
                isLoading={isLoading}
                setIsLoading={setIsLoading}
                stayLoggedIn={stayLoggedIn}
                setStayLoggedIn={setStayLoggedIn}
                errorMessage={errorMessage}
            />
        </div>
    )
}

export function LoginHeader() {
    return (
        <div className="login-intro">
            <Row className="justify-content-center">
                <Image rounded src={logo} height="200px" width="auto"/>
            </Row>
            <Row className="justify-content-center">
                <h1>Greetings Traveller!</h1>
            </Row>
            <Row className="justify-content-center">
                <h2>Be welcome at FileFighter</h2>
            </Row>
        </div>
    )
}

export function LoginInput(props: LoginInputInterface) {
    const {
        handleSubmit,
        username,
        setUsername,
        password,
        setPassword,
        isLoading,
        stayLoggedIn,
        setStayLoggedIn,
        errorMessage,
    } = props

    return (
        <div>
            <Row className="mt-4 justify-content-center">
                <Col className="login-input">
                    <Form onSubmit={handleSubmit}>
                        <Form.Group controlId="formBasicUsername">
                            <Form.Control placeholder="Username" value={username}
                                          onChange={event => setUsername(event.target.value)}/>
                        </Form.Group>

                        <Form.Group controlId="formBasicPassword">
                            <Form.Control type="password" placeholder="Password" value={password}
                                          onChange={event => setPassword(event.target.value)}/>
                            <Form.Text className="text-muted">
                                Contact your administrator if you have forgotten your login details.
                            </Form.Text>
                        </Form.Group>

                        <Button variant="primary" type="submit" block>
                            <Spinner
                                as="span"
                                animation="grow"
                                size="sm"
                                role="status"
                                aria-hidden="true"
                                className={isLoading ? "" : "d-none"}
                            /> <span className={isLoading ? "sr-only" : "d-none"}>isLoading...</span>Sign in
                        </Button>
                        <Form.Group controlId="formBasicCheckbox" className="mt-3 justify-content-center">
                            <Form.Check checked={stayLoggedIn} type="checkbox" label="Keep me signed in*"
                                        onChange={() => setStayLoggedIn(!stayLoggedIn)}/>
                            <Form.Text className="text-muted">
                                *By clicking this, you accept the usage of cookies.
                            </Form.Text>
                        </Form.Group>
                        <p className="text-danger">{errorMessage}</p>
                    </Form>
                </Col>
            </Row>
        </div>
    )
}

export default Login;
