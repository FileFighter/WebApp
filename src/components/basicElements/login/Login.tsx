import React, {
    Dispatch,
    FormEvent,
    ReactElement,
    SetStateAction,
    useEffect,
    useState,
} from "react"
import { Button, Col, Form, Image, Row, Spinner } from "react-bootstrap"
import { loginWithUsernameAndPassword } from "../../../background/api/auth"

import logo from "../../../assets/images/logos/logoWithWhiteBorder.png"
import { useLocation, useNavigate } from "react-router-dom"
import { useSelector } from "react-redux"
import { RootState } from "../../../background/redux/store"

export interface LoginInputInterface {
    handleSubmit: (event: FormEvent) => void
    username: string | number | string[] | undefined
    setUsername: Dispatch<SetStateAction<string>>
    password: string | number | string[] | undefined
    setPassword: Dispatch<SetStateAction<string>>
    isLoading: boolean
    setIsLoading: Dispatch<SetStateAction<boolean>>
    stayLoggedIn: boolean
    setStayLoggedIn: Dispatch<SetStateAction<boolean>>
    errorMessage: string | null
}

function Login(): ReactElement {
    const [userName, setUsername] = useState<string>("")
    const [password, setPassword] = useState<string>("")
    const [stayLoggedIn, setStayLoggedIn] = useState<boolean>(true)
    const [errorMessage, setErrorMessage] = useState<string>("")
    const [isLoading, setIsLoading] = useState<boolean>(false)

    const location = useLocation()
    const urlSearchParams = new URLSearchParams(location.search)
    const dest = urlSearchParams.get("dest")
    const navigate = useNavigate()

    const tokens = useSelector((state: RootState) => state.tokens)

    useEffect(() => {
        if (tokens.refreshToken && tokens.accessToken?.token) {
            if (dest) {
                navigate(decodeURIComponent(dest))
            } else {
                navigate("/")
            }
        }
    }, [dest, navigate, tokens])

    const handleSubmit = (event: FormEvent) => {
        event.preventDefault()
        setIsLoading(true)
        loginWithUsernameAndPassword(userName, password, stayLoggedIn)
            .then(() => {
                //nothing to do here :)
                setErrorMessage("")
                setIsLoading(false)
            })
            .catch((error) => {
                console.log(error)
                setIsLoading(false)
                setErrorMessage(error.response?.data.message)
            })
    }

    return (
        <LoginInteractionArea
            handleSubmit={handleSubmit}
            username={userName}
            setUsername={setUsername}
            password={password}
            setPassword={setPassword}
            isLoading={isLoading}
            setIsLoading={setIsLoading}
            stayLoggedIn={stayLoggedIn}
            setStayLoggedIn={setStayLoggedIn}
            errorMessage={errorMessage}
        />
    )
}

export function LoginInteractionArea(props: LoginInputInterface) {
    const {
        handleSubmit,
        username,
        setUsername,
        password,
        setPassword,
        isLoading,
        setIsLoading,
        stayLoggedIn,
        setStayLoggedIn,
        errorMessage,
    } = props
    return (
        <LoginInput
            handleSubmit={handleSubmit}
            username={username}
            setUsername={setUsername}
            password={password}
            setPassword={setPassword}
            isLoading={isLoading}
            setIsLoading={setIsLoading}
            stayLoggedIn={stayLoggedIn}
            setStayLoggedIn={setStayLoggedIn}
            errorMessage={errorMessage}
        />
    )
}

export function LoginHeader() {
    return (
        <div className="login-intro">
            <Row className="justify-content-center">
                <Image rounded src={logo} height="200px" width="auto" />
            </Row>
            <Row className="justify-content-center">
                <h1>Greetings Traveller!</h1>
            </Row>
            <Row className="justify-content-center">
                <h2>Be welcome at FileFighter</h2>
            </Row>
        </div>
    )
}

export function LoginInput(props: LoginInputInterface) {
    const {
        handleSubmit,
        username,
        setUsername,
        password,
        setPassword,
        isLoading,
        stayLoggedIn,
        setStayLoggedIn,
        errorMessage,
    } = props

    return (
        <div>
            <Row className="mt-4 justify-content-center">
                <Col className="login-input">
                    <Form onSubmit={handleSubmit}>
                        <Form.Group controlId="formBasicUsername" className="mb-3">
                            <Form.Control
                                autoFocus
                                placeholder="Username"
                                autoComplete="username"
                                value={username}
                                onChange={(event) =>
                                    setUsername(event.target.value)
                                }
                            />
                        </Form.Group>

                        <Form.Group controlId="formBasicPassword" className="mb-3">
                            <Form.Control
                                type="password"
                                placeholder="Password"
                                value={password}
                                autoComplete="current-password"
                                onChange={(event) =>
                                    setPassword(event.target.value)
                                }
                            />
                            <Form.Text className="text-muted">
                                Contact your administrator if you have forgotten
                                your login details.
                            </Form.Text>
                        </Form.Group>

                        <Button variant="primary" type="submit" className="w-100">
                            <Spinner
                                as="span"
                                animation="grow"
                                size="sm"
                                role="status"
                                aria-hidden="true"
                                className={isLoading ? "" : "d-none"}
                            />{" "}
                            <span className={isLoading ? "sr-only" : "d-none"}>
                                isLoading...
                            </span>
                            Sign in
                        </Button>
                        <Form.Group
                            controlId="formBasicCheckbox"
                            className="mt-3 justify-content-center"
                        >
                            <Form.Check
                                checked={stayLoggedIn}
                                type="checkbox"
                                label="Keep me signed in*"
                                onChange={() => setStayLoggedIn(!stayLoggedIn)}
                            />
                            <Form.Text className="text-muted">
                                *By clicking this, you accept the usage of
                                cookies.
                            </Form.Text>
                        </Form.Group>
                        <p className="text-danger">{errorMessage}</p>
                    </Form>
                </Col>
            </Row>
        </div>
    )
}

export default Login
