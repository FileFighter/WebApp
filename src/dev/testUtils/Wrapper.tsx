import React from "react";
import { BrowserRouter } from "react-router-dom";
import store from "../../background/redux/store";
import { Provider } from "react-redux";

export const Wrapper: React.FC = ({ children }) => {
    return (
        <Provider store={store}>
            <BrowserRouter>{children}</BrowserRouter>
        </Provider>
    );
};
