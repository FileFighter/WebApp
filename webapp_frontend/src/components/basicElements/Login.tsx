import React, {FormEvent, ReactElement, useState} from 'react';
import {Button, Form, Container, Row,Col,Spinner} from "react-bootstrap";
import {loginWithUsernameAndPassword} from "../../background/api/auth";

function Login(): ReactElement {
    const [userName, setUsername] = useState<string>("");
    const [password, setPassword] = useState<string>("");
    const [stayLoggedIn, setStayLoggedIn] = useState<boolean>(true);
    const [errorMessage, setErrorMessage] = useState<string>("");
    const [loading,setLoading]=useState<boolean>(false);


    const handleSubmit = (event: FormEvent) => {
        event.preventDefault();
        setLoading(true)
        loginWithUsernameAndPassword(userName, password,stayLoggedIn)
            .then(() => {
                //nothing to do here :)
                setErrorMessage("");
                setLoading(false);
            })
            .catch(((error) => {
                console.log(error);
                setLoading(false)
                setErrorMessage(error.response?.data.message);
            }))


    };


    return (
        <Container>
            <Row >
                <Col md={{ span: 6, offset: 3 }}>
                    <Form onSubmit={handleSubmit}>
                        <Form.Group controlId="formBasicEmail">
                            <Form.Label>Username</Form.Label>
                            <Form.Control placeholder="Enter username" value={userName} onChange={event => setUsername(event.target.value)}/>
                        </Form.Group>

                        <Form.Group controlId="formBasicPassword">
                            <Form.Label>Password</Form.Label>
                            <Form.Control type="password" placeholder="Password" value={password} onChange={event => setPassword(event.target.value)}/>
                            <Form.Text className="text-muted">
                                Contact your administrator if you have forgotten your password.
                            </Form.Text>
                        </Form.Group>
                        <Form.Group controlId="formBasicCheckbox">
                            <Form.Check checked={true} type="checkbox" label="Stay logged in. (By clicking this you accept the usage of cookies.)" onChange={() => setStayLoggedIn(!stayLoggedIn)}/>
                        </Form.Group>
                        <Button variant="primary" type="submit" >
                            <Spinner
                                as="span"
                                animation="grow"
                                size="sm"
                                role="status"
                                aria-hidden="true"
                                className={loading? "" :"d-none"}
                            />   <span className={loading? "sr-only" :"d-none"}>Loading...</span>Submit
                        </Button>

                        <p className="text-danger">{errorMessage}</p>
                    </Form>
                </Col>
            </Row>
        </Container>);
}

export default Login;
