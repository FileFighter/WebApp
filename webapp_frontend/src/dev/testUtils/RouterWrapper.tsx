import React from "react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const RouterWrapper: React.FC = ({ children }) => {
  const history = createMemoryHistory();

  return <Router history={history}> {children} </Router>;
};
