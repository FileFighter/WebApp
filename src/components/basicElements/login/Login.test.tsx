import { render } from "@testing-library/react";
import React from "react";
import Login from "./Login";
import {RouterWrapper} from "../../../dev/testUtils/RouterWrapper";

test("renders Login page", () => {
    const breadcrumb = render(
        <RouterWrapper>
            <Login />
        </RouterWrapper>
    );

    expect(breadcrumb).toMatchSnapshot();
});