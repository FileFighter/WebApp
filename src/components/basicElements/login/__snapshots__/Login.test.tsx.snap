// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="mt-4 justify-content-center row"
        >
          <div
            class="login-input col"
          >
            <form
              class=""
            >
              <div
                class="mb-3"
              >
                <input
                  autocomplete="username"
                  class="form-control"
                  id="formBasicUsername"
                  placeholder="Username"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <input
                  autocomplete="current-password"
                  class="form-control"
                  id="formBasicPassword"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <small
                  class="text-muted form-text"
                >
                  Contact your administrator if you have forgotten your login details.
                </small>
              </div>
              <button
                class="w-100 btn btn-primary"
                type="submit"
              >
                <span
                  aria-hidden="true"
                  class="d-none spinner-grow spinner-grow-sm"
                  role="status"
                />
                 
                <span
                  class="d-none"
                >
                  isLoading...
                </span>
                Sign in
              </button>
              <div
                class="mt-3 justify-content-center"
              >
                <div
                  class="form-check"
                >
                  <input
                    checked=""
                    class="form-check-input"
                    id="formBasicCheckbox"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="formBasicCheckbox"
                    title=""
                  >
                    Keep me signed in*
                  </label>
                </div>
                <small
                  class="text-muted form-text"
                >
                  *By clicking this, you accept the usage of cookies.
                </small>
              </div>
              <p
                class="text-danger"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="mt-4 justify-content-center row"
      >
        <div
          class="login-input col"
        >
          <form
            class=""
          >
            <div
              class="mb-3"
            >
              <input
                autocomplete="username"
                class="form-control"
                id="formBasicUsername"
                placeholder="Username"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <input
                autocomplete="current-password"
                class="form-control"
                id="formBasicPassword"
                placeholder="Password"
                type="password"
                value=""
              />
              <small
                class="text-muted form-text"
              >
                Contact your administrator if you have forgotten your login details.
              </small>
            </div>
            <button
              class="w-100 btn btn-primary"
              type="submit"
            >
              <span
                aria-hidden="true"
                class="d-none spinner-grow spinner-grow-sm"
                role="status"
              />
               
              <span
                class="d-none"
              >
                isLoading...
              </span>
              Sign in
            </button>
            <div
              class="mt-3 justify-content-center"
            >
              <div
                class="form-check"
              >
                <input
                  checked=""
                  class="form-check-input"
                  id="formBasicCheckbox"
                  type="checkbox"
                />
                <label
                  class="form-check-label"
                  for="formBasicCheckbox"
                  title=""
                >
                  Keep me signed in*
                </label>
              </div>
              <small
                class="text-muted form-text"
              >
                *By clicking this, you accept the usage of cookies.
              </small>
            </div>
            <p
              class="text-danger"
            />
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
